Formula/spago.rb

```diff
@@ -0,0 +1,65 @@
+class Spago < Formula
+  desc "PureScript package manager and build tool"
+  homepage "https://github.com/purescript/spago"
+  url "https://github.com/purescript/spago/archive/refs/tags/0.20.4.tar.gz"
+  sha256 "e2ef8604115556b39ec71301d85a3502502fd5972e40bac82cd556d6a128baff"
+  license "BSD-3-Clause"
+  head "https://github.com/purescript/spago.git", branch: "master"
+
+  depends_on "ghc" => :build
+  depends_on "haskell-stack" => :build
+  depends_on "purescript"
+
+  resource "docs-search-app-0.0.10.js" do
+    url "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.10/docs-search-app.js"
+    sha256 "45dd227a2139e965bedc33417a895ec7cb267ae4a2c314e6071924d19380aa54"
+  end
+
+  resource "docs-search-app-0.0.11.js" do
+    url "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/docs-search-app.js"
+    sha256 "0254c9bd09924352f1571642bf0da588aa9bdb1f343f16d464263dd79b7e169f"
+  end
+
+  resource "purescript-docs-search-0.0.10" do
+    url "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.10/purescript-docs-search"
+    sha256 "437ac8b15cf12c4f584736a07560ffd13f4440cd0c44c3a6f7a29248a1ff8171"
+  end
+
+  resource "purescript-docs-search-0.0.11" do
+    url "https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/purescript-docs-search"
+    sha256 "06dfcb9b84408527a2980802108fae6a5260a522013f67d0ef7e83946abe4dc2"
+  end
+
+  def install
+    # Equivalent to make fetch-templates:
+    (buildpath/"templates").cd do |d|
+      d.install resource("docs-search-app-0.0.10.js"), resource("purescript-docs-search-0.0.10")
+      (d/"docs-search-app.js").rename "docs-search-app-0.0.10.js"
+      (d/"purescript-docs-search").rename "purescript-docs-search-0.0.10"
+      d.install resource("docs-search-app-0.0.11.js"), resource("purescript-docs-search-0.0.11")
+      (d/"docs-search-app.js").rename "docs-search-app-0.0.11.js"
+      (d/"purescript-docs-search").rename "purescript-docs-search-0.0.11"
+    end
+
+    system "stack", "install", "--system-ghc", "--no-install-ghc", "--skip-ghc-check", "--local-bin-path=#{bin}"
+    (bash_completion/"spago").write `#{bin}/spago --bash-completion-script #{bin}/spago`
+    (zsh_completion/"_spago").write `#{bin}/spago --zsh-completion-script #{bin}/spago`
+  end
+
+  test do
+    assert_match "Spago - manage your PureScript projects", shell_output("#{bin}/spago --help")
+    testpath.cd do
+      system bin/"spago", "init"
+    end
```

Comment:
```suggestion
    system bin/"spago", "init"
```
We can drop the `cd`s, since `testpath` is the current working directory.